docs/howto/danceable-lyrics-input-tool/README.md

```diff
@@ -0,0 +1,485 @@
+        # truncated code
+
+    def on_record_batch(self, batch: "Table", anchor: namedtuple) -> None:
+        # truncated code
+
+    def on_complete(self) -> None:
+        import pandas as pd
+        import pyarrow as pa
+
+        df = pd.DataFrame(
+            {
+                "x": [1, 2, 3],
+                "y": ["hello", "world", "from ayx_python_sdk!"],
+                "z": [self.config_value, self.config_value, self.config_value],
+            }
+        )
+
+        packet = pa.Table.from_pandas(df)
+
+        self.provider.write_to_anchor("Output", packet)
+        self.provider.io.info("DanceableLyrics tool done.")
+```
+
+# Write a Plugin!
+In this guide, we will be utilzing multiple data files, reading each directly illustrating a "connector" like plugin. That is, to connect directly to local or remote data sources. While we will be reading CSV input files, these could be in any format including those not directly supported by Designer!
+
+> :information_source: To see an Input tool connecting to a remote API, check out [Creating an API Tool](https://github.com/alteryx/ayx-developer-sdk/blob/main/docs/howto/how-to-make-api-tool/api-tool-guide.md)!
+
+> :information_source: Since this is an Input tool, we only focus on the `__init__` and `on_complete` functions. For additional information on the lifecycle of a plugin, refer to the [AYX Python SDK documentation](https://alteryx.github.io/ayx-python-sdk/plugin_lifecycle.html).
+
+## 1. Dependencies
+The first thing we'll want to do is update our `./backend/requirements-thirdparty.txt` telling Pything we depend on Polars as a third party depencency. This is as easy as adding a single line to the file:
+
+```txt
+polars
+```
+
+## 2. Imports
+The next step is update our Python module's imports. Imports tell Python what other code we will be referencing in our plugin. Head near the top of your `danceable_lyrics.py` and add the required import statements. Your code should look like this:
+
+```python
+from collections import namedtuple
+
+from ayx_python_sdk.core import PluginV2
+from ayx_python_sdk.core.field import FieldType
+from ayx_python_sdk.core.utils import create_schema
+from ayx_python_sdk.providers.amp_provider.amp_provider_v2 import AMPProviderV2
+
+import polars as pl
+import pyarrow as pa
+```
+
+## 3. Initialization
+Let's now update the plugin's `__init__` method, setting up some basic variables we will reference later in the code. To start, modify the start of `__init__` to set a base path to **where your data assets are extracted to** using `self.DATASETS_BASE`. Below is what my copy looks like:
+
+```python
+def __init__(self, provider: AMPProviderV2) -> None:
+    """Construct a plugin."""
+    self.provider = provider
+    self.name = "DanceableLyrics"
+
+    self.DATASETS_BASE = "c:/users/bryan.ashby/DanceableLyricsData/"
+```
+
+Next, let's tell downstream tools what our plugin's metadata looks like. That is, what fields and their types will we be outputting:
```

Comment:
TW review:

Next, let's tell downstream tools what our plugin's metadata looks like. That is, what fields and their data types the plugin will output: